src/meta/src/stream/scale.rs

```diff
@@ -897,11 +1169,225 @@ where
                     (None, None) => {
                         // do nothing
                     }
-                    _ => bail!("vnode mapping update for scaling is not supported for now"),
+                    _ => {
+                        if let Some(downstream_updated_bitmap) =
+                            updated_bitmap.get(&downstream_fragment_id)
+                        {
+                            // if downstream scale in/out
+                            let mut raw = vec![0 as ActorId; VIRTUAL_NODE_COUNT];
+
+                            for (actor_id, bitmap) in downstream_updated_bitmap {
+                                for (idx, pos) in raw.iter_mut().enumerate() {
+                                    if bitmap.is_set(idx).unwrap() {
+                                        *pos = *actor_id;
+                                    }
+                                }
+                            }
+
+                            let (original_indices, data) = compress_data(&raw);
```

Comment:
Looks like a little bit duplicated with below. Can we extract it? (Or we may introduce some specific structs for mapping, bitmap stuffs in the future)
https://github.com/risingwavelabs/risingwave/blob/752dd2b70fbf48ac8a1cb24379a4fa0a07aff4cd/src/meta/src/stream/scale.rs#L870-L892